### DISCUSSION TITLE
What is GITHUB_ACCESS_TOKEN in File based Storage?

### DISCUSSION THREAD

**Discussion by Srj** [PrefectHQ/prefect#3794] [Q&A] [answered]
```
# flows/my_flow.py

from prefect import task, Flow
from prefect.environments.storage import GitHub

@task
def get_data():
    return [1, 2, 3, 4, 5]

@task
def print_data(data):
    print(data)

with Flow("file-based-flow") as flow:
    data = get_data()
    print_data(data)

flow.storage = GitHub(
    repo="org/repo",                 # name of repo
    path="flows/my_flow.py",        # location of flow file in repo
    secrets=["GITHUB_ACCESS_TOKEN"]  # name of personal access token secret  !!!!!!!WHAT IS THISSSSSSS!!!!!!
)
```

### What should i pass here? I have copied my access token for github here. It fails. The Error Message in UI is
`Failed to load and execute Flow's environment: ClientError('400 Client Error: Bad Request for url: http://localhost:4200/graphql\n\nThe following error messages were provided by the GraphQL server:\n\n GRAPHQL_VALIDATION_FAILED: Cannot query field "secret_value" on type "Query".\n\nThe GraphQL query was:\n\n query($name: String!) {\n secret_value(name: $name)\n }\n\nThe passed variables were:\n\n {"name": "0f*************************************"}\n')`

**Comment by joshmeek** [comment, marked as answer]
Hi @Srj the `GITHUB_ACCESS_TOKEN` is one of the default secrets that you can configure to automatically authenticate with github. See more information [here](https://docs.prefect.io/core/concepts/secrets.html#default-secrets) for default secret and [here](https://docs.prefect.io/core/concepts/secrets.html) for more info on secrets as a whole.

**Reply by Srj** [reply]
So, according to documentation, all i have to do is, set a environment variable 
PREFECT__CONTEXT__SECRETS__GITHUB_ACCESS_TOKEN={MY TOKEN STRING}              ##Please Check the syntax
and make no change in the code above.
That's all?

**Reply by joshmeek** [reply]
Yeah that appear correct. However that value needs to be present where the flow run starts. So you may need to provide an environment variable to your agent through `--env` https://docs.prefect.io/api/latest/cli/agent.html

**Reply by Srj** [reply]
Thank You. It worked. Needed to pass --env to agent.
